app/src/main/java/ch/epfl/sweng/SDP/auth/AccountCreationActivity.java

```diff
@@ -1,31 +1,69 @@
 package ch.epfl.sweng.SDP.auth;
 
-import static android.preference.PreferenceManager.getDefaultSharedPreferences;
-
+import android.content.Intent;
 import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
 import android.view.View;
-import ch.epfl.sweng.SDP.Activity;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import ch.epfl.sweng.SDP.Account;
+import ch.epfl.sweng.SDP.Constants;
 import ch.epfl.sweng.SDP.R;
 import ch.epfl.sweng.SDP.home.HomeActivity;
 
-public class AccountCreationActivity extends Activity {
+public class AccountCreationActivity extends AppCompatActivity {
+    private EditText usernameInput;
+    private Button createAcc;
+    private TextView usernameTaken;
+    private String username;
+    private Account account;
+    private View.OnClickListener createAccListener;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_account_creation);
+        usernameInput = this.findViewById(R.id.usernameInput);
+        createAcc = this.findViewById(R.id.createAcc);
+        createAcc.setOnClickListener(createAccListener);
+        usernameTaken = this.findViewById(R.id.usernameTaken);
+        createAccListener = new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                createAccClicked();
+            }
+        };
+        createAcc.setOnClickListener(createAccListener);
+    }
+
+    /**
+     * Gets called when user entered username and clicked on create account.
+     */
+    public void createAccClicked() {
+        username = usernameInput.getText().toString();
+        if (username.isEmpty()) {
+            usernameTaken.setText("Username must not be empty.");
+        } else {
+            account = new Account(new Constants(), username);
+            try{
+                account.checkIfAccountNameIsFree(username);
+                account.registerAccount(this.getApplicationContext());
+                gotoHome();
+            } catch (Exception exception){
+                usernameTaken.setText(exception.getMessage());
+            }
+        }
     }
 
     /**
-     * Create an account and update the database.
-     *
-     * @param view the button clicked
+     * Calls HomeActivity.
      */
-    public void createAccount(View view) {
-        getDefaultSharedPreferences(getApplicationContext()).edit().putBoolean("hasAccount", true)
```

Comment:
This line is needed otherwise the login does not work as expected; it should be moved inside createAccClicked

Reply:
That line is used in the Main too, yes you should put it back